Source/Core/DolphinQt/CheatsManager.cpp

```diff
@@ -175,11 +175,12 @@ void CheatsManager::OnStateChanged(Core::State state)
   if (state != Core::State::Running && state != Core::State::Paused)
     return;
 
-  auto* model = Settings::Instance().GetGameListModel();
+  if (!m_game_list_model)
```

Comment:
I should have asked before.
In what situation might `m_game_list_model` be null here?
If that's actually possible it might deserve an explanation in a comment.

Reply:
Actually, this is a good question. Since they were pointers, I felt bound to check that the pointer is not null. It seems we can use const references instead.

There is no real reason to have the GameListModel dynamically allocated in the GameList constructor; it can simply be an object. Then, instead of sharing pointers to the object, const references can be used. By using const references, we are asserting that the GameListModel will outlive the components using it, which is always the case.

I will be adding a new commit for this, but it can be squashed later.